src/main/c/jni/org_jpy_PyLib.c

```
@@ -1217,7 +1221,10 @@ JNIEXPORT jlong JNICALL Java_org_jpy_PyLib_getLongValue
 
     pyObject = (PyObject*) objId;
     value = JPy_AS_CLONG(pyObject);
-
+    if (value == -1 && PyErr_Occurred()) {
+        JPy_DIAG_PRINT(JPy_DIAG_F_ALL, "Java_org_jpy_PyLib_getLongValue: error: failed to convert Python object to Java long\n");
+        PyLib_HandlePythonException(jenv);
+    }
```

I think we want to consider other exceptional cases - for example, when the value overflows, it throws an exception instead of returning an error value: https://docs.python.org/3/c-api/long.html#c.PyLong_AsLong

---

I don't think the Overflow exception and returning -1 are mutually exclusive. That condition check is a common pattern even in the cpython source.